File: Baekjoon/2491.py
import sys
from bisect import bisect_left

n = int(sys.stdin.readline().rstrip())
num = list(map(int, sys.stdin.readline().rstrip().split()))


def two_pointer():
    left = 0
    right = 0
    last_value = 0
    ret = 0
    while True:
        if left == right:
            last_value = num[right]
            right += 1
            ret = max(ret, right - left)
        elif right == n:
            break
        elif num[right] >= last_value:
            last_value = num[right]
            right += 1
            ret = max(ret, right - left)
        else:
            left += 1
    return ret


answer = two_pointer()
num.reverse()
answer = max(answer, two_pointer())
print(answer)

File: Baekjoon/2491.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int n;
vector<int> num;

int two_pointer() {
	int left = 0;
	int right = 0;
	int last_value = 0;
	int ret = 0;

	while (true) {
		if (left == right) {
			last_value = num[right];
			right++;
			ret = max(ret, right - left);
		}
		else if (right == n) break;
		else if (num[right] >= last_value) {
			last_value = num[right];
			right++;
			ret= max(ret, right - left);
		}
		else left++;
	}
	return ret;
}

int main() {
	ios_base::sync_with_stdio(0);
	cin.tie(0);

	cin >> n;
	num = vector<int>(n);
	for (int i = 0; i < n; i++)
		cin >> num[i];
	int answer = two_pointer();
	reverse(num.begin(), num.end());
	answer = max(answer, two_pointer());
	cout << answer << '\n';
	return 0;
}